src/quafu/tasks/tasks.py

```diff
@@ -248,7 +248,12 @@ def send(self,
                 else:
                     self.submit_history[group].append(task_id)
 
-                return ExecResult(res_dict)
+                result = ExecResult(res_dict)
+                while wait and (self.shots > 0) and (not result.counts):
+                    if verbose:
+                        logging.warning('warning for quafu status: empty result.counts, retrieving task...')
+                    result = self.retrieve(result.taskid)
```

Comment:
this will result in Deadpool if retrieve result counts is also none. suggest to add some limit